src/clr-angular/utils/drag-and-drop/draggable.ts

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+    AfterViewInit,
+    ComponentFactory,
+    ComponentFactoryResolver,
+    ContentChild,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Inject,
+    Injector,
+    Input,
+    OnDestroy,
+    Output,
+    Renderer2,
+    ViewContainerRef
+} from "@angular/core";
+import {Subscription} from "rxjs/Subscription";
+
+import {DomAdapter} from "../../data/datagrid/render/dom-adapter";
+
+import {DraggableGhost} from "./draggable-ghost";
+import {DraggableHandle} from "./draggable-handle";
+import {IfDragged} from "./if-dragged";
+import {DragPosition} from "./interfaces/client-position";
+import {DraggableEvent} from "./interfaces/draggable-event";
+import {DragAndDropDispatcher} from "./providers/drag-and-drop-dispatcher";
+import {DragDispatcher} from "./providers/drag-dispatcher";
+import {CUSTOM_GHOST_STATE, CUSTOM_GHOST_STATE_PROVIDER, CustomGhostState} from "./providers/draggable-tracker";
+
+@Directive({selector: "[clrDraggable]", providers: [DragDispatcher, DomAdapter, CUSTOM_GHOST_STATE_PROVIDER]})
+export class Draggable implements AfterViewInit, OnDestroy {
+    constructor(private dragDispatcher: DragDispatcher, private dragAndDrop: DragAndDropDispatcher,
+                private domAdapter: DomAdapter, private renderer: Renderer2, private el: ElementRef,
+                private viewContainerRef: ViewContainerRef, private componentFactoryResolver: ComponentFactoryResolver,
+                private injector: Injector, @Inject(CUSTOM_GHOST_STATE) private customGhostState: CustomGhostState) {
+        this.componentFactory = this.componentFactoryResolver.resolveComponentFactory(DraggableGhost);
+    }
+
+    private componentFactory: ComponentFactory<DraggableGhost>;
+
+    private clientLeft: number;
+    private clientTop: number;
+
+    private initMoveX: number;
+    private initMoveY: number;
+
+    private initMoveXOnEl: number;
+    private initMoveYOnEl: number;
+
+    private isDragMoveStarted: boolean = false;
+
+    @Output("clrDragStart") private dragStartedEmitter: EventEmitter<DraggableEvent> = new EventEmitter();
+    @Output("clrDragMove") private dragMovedEmitter: EventEmitter<DraggableEvent> = new EventEmitter();
+    @Output("clrDragEnd") private dragEndedEmitter: EventEmitter<DraggableEvent> = new EventEmitter();
+
+    // This input is or transfering payload from draggable to dropppable
+    // We probably need to have a definite type on it with an interface instead of having any.
+    @Input("clrDraggable") data: any;
+
+    @Input("clrDragAndDropGroup") groupKey: string;
```

Comment:
Yup, there it is. Draggables can belong to more than one group, so this is `string | string[]` too.